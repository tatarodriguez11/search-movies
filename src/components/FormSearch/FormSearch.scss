.search__form-sort--container {
  display: flex;
  align-items: center;
  justify-content: center;
}

.search__form-container, .search__form-sort--container{
  border: none;
}

.search__form-input {
  margin: 0;
  width: 75%;
  display: inline;
  border: 1px solid transparent;
}

.search__form-btn {
  margin-left: 2em;
}

.search__form-error{
  color: red;
}