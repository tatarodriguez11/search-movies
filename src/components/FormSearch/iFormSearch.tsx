export interface FormSearchProps {
  handleChange: (event: any) => void, 
  handleSubmit: (event: any) => void, 
  handleSort: () => void, 
  sort: boolean, 
  error: string
  search?: any,
}