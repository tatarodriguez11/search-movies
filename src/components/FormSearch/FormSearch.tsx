import { FormSearchProps } from "./iFormSearch";
import './FormSearch.scss'
export function FormSearch ({handleChange, handleSubmit, handleSort, sort, search,error }: FormSearchProps) {

  return(
    <header>
      <form name="form" className="search__form" onSubmit={handleSubmit}>
        <fieldset className="search__form-container">
          <input
            className="search__form-input"
            style={{
              borderColor: error ? "red" : "transparent",
            }}
            onChange={handleChange}
            value={search}
            name="query" 
            data-testid="search-input"
            placeholder="Avengers, Star Wars, The Matrix..."
          />
          <button className="search__form-btn" type="submit">
            Buscar
          </button>
        </fieldset>
        <fieldset className="search__form-sort--container">
          <input
            id="sort"
            type="checkbox"
            onChange={handleSort}
            checked={sort}
          />
          <label htmlFor="sort">Ordenar resultados</label>
        </fieldset>
      </form>
      {error && <p className="search__form-error">{error}</p>}
      <hr />
    </header>
  )
}