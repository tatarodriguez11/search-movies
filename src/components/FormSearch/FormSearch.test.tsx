import { render, fireEvent, act } from "@testing-library/react";
import { FormSearch } from "./FormSearch";

describe("FormSearch", () => {
  const props = {
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    handleSort: jest.fn(),
    sort: false,
    error: "",
    search: "",
  };

  it("matches the snapshot", () => {
    const { container } = render(<FormSearch {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("displays an error when provided", () => {
    const testError = "Test error message";
    const { getByText } = render(<FormSearch {...props} error={testError} />);
    expect(getByText(testError)).toBeInTheDocument();
  });

  it("calls handleSubmit on form submit", () => {
    const onSubmit = jest.fn().mockImplementation((event) => {
      event.preventDefault();
    });
    const { getByText } = render(
      <FormSearch {...props} handleSubmit={onSubmit} />
    );
    const button = getByText("Buscar");

    fireEvent.click(button);
    expect(onSubmit).toHaveBeenCalledTimes(1);
  });

  it("calls handleChange when input value changes", () => {
    const handleChange = jest.fn();
    const { getByTestId } = render(
      <FormSearch {...props} handleChange={handleChange} />
    );
    const input = getByTestId("search-input");
    act(() => {
      fireEvent.change(input, { target: { value: "a" } });
    });

    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it("calls handleSort on checkbox click", () => {
    const handleSort = jest.fn().mockImplementation((event) => {
      event.preventDefault();
    });
    const { getByLabelText } = render(
      <FormSearch {...props} handleSort={handleSort} />
    );
    const checkbox = getByLabelText("Ordenar resultados");
    act(() => {
      fireEvent.click(checkbox);
    });
    expect(handleSort).toHaveBeenCalledTimes(1);
  });

  it("renders the correct placeholder text", () => {
    const { getByPlaceholderText } = render(<FormSearch {...props} />);
    const input = getByPlaceholderText("Avengers, Star Wars, The Matrix...");
    expect(input).toBeInTheDocument();
  });
});
