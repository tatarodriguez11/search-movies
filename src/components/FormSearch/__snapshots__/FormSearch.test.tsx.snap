// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSearch matches the snapshot 1`] = `
<div>
  <header>
    <form
      class="search__form"
      name="form"
    >
      <fieldset
        class="search__form-container"
      >
        <input
          class="search__form-input"
          data-testid="search-input"
          name="query"
          placeholder="Avengers, Star Wars, The Matrix..."
          style="border-color: transparent;"
          value=""
        />
        <button
          class="search__form-btn"
          type="submit"
        >
          Buscar
        </button>
      </fieldset>
      <fieldset
        class="search__form-sort--container"
      >
        <input
          id="sort"
          type="checkbox"
        />
        <label
          for="sort"
        >
          Ordenar resultados
        </label>
      </fieldset>
    </form>
    <hr />
  </header>
</div>
`;
